src/react-imgix.jsx

```diff
@@ -117,43 +111,44 @@ function buildSrc({
     srcSet = src;
   } else {
     if (fixedSize) {
-      const { q, ...urlParams } = srcOptions;
-      const constructedUrl = constructUrl(rawSrc, urlParams);
-
-      let srcFn = buildDprSrcWithQualityByDpr;
+      const { width, w, height, h, q, ...urlParams } = srcOptions;
```

Comment:
No change needed, but I think it would be an improvement if these width and w values were "collapsed" down to a single value before getting to this point. IMO having both here seems like a bit of a smell

Reply:
I agree, what would you suggest?

Reply:
I would suggest having a "props processing" stage that occurs at the start of the component render function. This stage would take the component props (or at first it could just take the imgix params), and transform this down to a consistent set of params with all aliases flattened to the un-aliased values. For example:

```jsx
const ReactImgix = props => {
	// Process imgix params
	const imgixParams = sanitizeImgixParams(props.imgixParams);

	// Now in the rest of the function, these imgix params can be passed around safely
	const src = buildUrl(path, imgixParams);

	return <img src={src} />;
}


const ALIAS_MAP = {width: 'w', height: 'h', /* ... */ };
const sanitizeImgixParams = imgixParams => Object.entries(imgixParams).reduce((acc, [key, value]) => {
	const targetKey = key in ALIAS_MAP ? ALIAS_MAP[key] : key;
	acc[targetKey] = value;
	return acc
}, {});
```

This naive implementation doesn't make any guarantees about the order that the values will be un-aliased/applied, but I think that's ok.

A more in-depth implementation here could handle the interactions between the imgix params that are currently able to be set with props and also inside `imgixParams`.

Reply:
Yeah, I think we really need to do some work to improve the hygiene around props, width, height, etc. There's the issue of duplicate params in Background URLs, plus there's the issues above wrt sizes, so this could all use some work, I agree.